### PR TITLE
new package: py-textblob

### PR DESCRIPTION
**Pull request by aweits**
add variant to py-nltk to allow for data download/installation
add dependencies to py-nltk so that bin/nltk works

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-nltk/package.py [hunk 1]
```diff
@@ -13,11 +13,28 @@ class PyNltk(PythonPackage):
...
+
+    @run_after('install')
+    def install_data(self):
+        if '+data' in self.spec:
+            mkdirp(prefix.data)
+            python('-m',
+                   'nltk.downloader',
+                   '-d',
+                   prefix.data,
+                   'all')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nltk/package.py]: I assume this requires internet access? I wonder if there is a way we could add the data as resources.

File: var/spack/repos/builtin/packages/py-nltk/package.py [hunk 2]
```diff
@@ -13,11 +13,28 @@ class PyNltk(PythonPackage):
 
     version('3.5', sha256='845365449cd8c5f9731f7cb9f8bd6fd0767553b9d53af9eb1b3abf7700936b35')
 
+    variant('data', default=False, description='Download the NLTK data')
+
     depends_on('python@3.5:', type=('build', 'run'))
-    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools', type=('build', 'run'))
     depends_on('py-click', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nltk/package.py:21]: It looks like we're also missing a dependency on `py-joblib`.

File: var/spack/repos/builtin/packages/py-nltk/package.py [hunk 3]
```diff
@@ -13,11 +13,28 @@ class PyNltk(PythonPackage):
...
     depends_on('py-regex', type=('build', 'run'))
     depends_on('py-tqdm', type=('build', 'run'))
+    depends_on('py-importlib-metadata', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nltk/package.py]: I wonder why they don't document this. I also don't see it used by any files in the tarball.

File: var/spack/repos/builtin/packages/py-nltk/package.py [hunk 4]
```diff
@@ -13,11 +13,28 @@ class PyNltk(PythonPackage):
...
+
+    def setup_run_environment(self, run_env):
+        if '+data' in self.spec:
+            run_env.prepend_path("NLTK_DATA", self.prefix.data)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-nltk/package.py]:
```suggestion
    def setup_run_environment(self, env):
        if '+data' in self.spec:
            env.prepend_path("NLTK_DATA", self.prefix.data)
```

File: var/spack/repos/builtin/packages/py-nltk/package.py (hunk 1 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-nltk/package.py]: Gave that a go.

File: var/spack/repos/builtin/packages/py-nltk/package.py (hunk 3 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-nltk/package.py]: This is generated by the console_scripts stuff, but setuptools on it's own suffices.

File: var/spack/repos/builtin/packages/py-nltk/package.py (hunk 2 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-nltk/package.py:21]: added

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**